python/pyspark/sql/dataframe.py

```diff
@@ -784,7 +824,7 @@ def foreach(self, f):
         """
         self.rdd.foreach(f)
 
-    def foreachPartition(self, f):
+    def foreachPartition(self, f: Callable[[Iterator[Row]], None]) -> None:
```

Comment:
Shouldn't this be `Iterable[Row]` instead of `Iterator[Row]`, so that it matches https://github.com/apache/spark/blob/master/python/pyspark/rdd.py#L1750 ?